### PR TITLE
Vali flail

### PR DESCRIPTION
**Pull request by tanker4390**
## About The Pull Request
A new melee weapon for use with the Vali system. This one has a reach of 3 for maximum range.
Deals 60 damage when wielded, with 1.5s attack delay.
Alternate attack will hit 3 tiles next to the target perpendicularly to the user, and stagger slow them. Cooldown in 6 seconds.

Screenshot 2023-05-30 230637

Screenshot 2023-05-30 230519

Screenshot 2023-05-30 230540
## Why It's Good For The Game
More loadout choices and playstyles.
## Changelog
:cl: Tanker
add: Added the Harvester flail for use with the Vali system.
/:cl:

### CODE REVIEW

**Review by Kuro020 — CHANGES_REQUESTED** [review]
>vali flail
>is a vali sword on a cord

uhhhhhhhhhhhhhhhhhhhhhhhhhh, yeah this needs better sprites, if maints approve the idea I'll sprite something for it

**Comment by tanker4390** [comment]
> uhhhhhhhhhhhhhhhhhhhhhhhhhh, yeah this needs better sprites, if maints approve the idea I'll sprite something for it

Yeah sorry I was a bit lazy with the sprites, not to mention that I don't know where I was going sprite-wise with "harvester flail"

**Comment by Pariah919** [comment]
I feel like one day we'll just have a vali gun at this point.

**Comment by bowlofcereal** [comment]
thats not a flail 💀 

https://en.wikipedia.org/wiki/Kusarigama

**Comment by Lumipharon** [comment]
Not really a fan of 'vali spear +'

**Comment by blackdav123** [comment]
there are a lot more interesting ways you can make a weapon like this instead of just "vali with 3 range this time"

give it a sweeping attack like ravage or something

**Comment by tanker4390** [comment]
> [vali spear II, rework]

Okey, I'll see how I can rework it... maybe the other weapons can have their own special attacks, too? I'll see to it

**Comment by chizzy376** [comment]
Do we really need another longer tile reach vali weapon?

**Comment by theta891** [comment]
more vali weapons is based

a shame its not a vali claw or fist

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -581,3 +581,102 @@
...
+	force_wielded = 60
+	throwforce = 25
+	edge = 1
+	reach = 3
+	attack_speed = 1.5 SECONDS
+	sharp = IS_SHARP_ITEM_SIMPLE
+	w_class = WEIGHT_CLASS_BULKY
+	flags_item = TWOHANDED
+	COOLDOWN_DECLARE(harvester_special_attack_cooldown)
+	var/obj/effect/abstract/particle_holder/particle_holder
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:600]: autodoc

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -581,3 +581,102 @@
...
+		var/mob/living/attacking = ravaged
+		if(attacking.stat == DEAD)
+			continue
+		step_away(attacking, H, 1, 2)
+		attacking.attacked_by(src, H, H.zone_selected)
+		shake_camera(attacking, 1, 1)
+		attacking.adjust_stagger(1)
+		attacking.adjust_slowdown(1)
+
+/obj/item/weapon/twohanded/flail/proc/activate_particles(mob/living/user, direction)
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/twohanded.dm:652]: autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest is whatever imo

**Comment by ClosetedSkeletonGH** [comment]
The amount of melee marines I've seen recently are ridiculous. The amount of melee tools there are is ridiculous. The more melee tools there are, the more melee marines there will be, the more marines will just throw. Getting stared at by 4 marines who dont bring a gun isnt fun and I feel like more vali weapons just encourage it.

**Comment by tanker4390** [comment]
> The amount of melee marines I've seen recently are ridiculous. The amount of melee tools there are is ridiculous. The more melee tools there are, the more melee marines there will be, the more marines will just throw. Getting stared at by 4 marines who dont bring a gun isnt fun and I feel like more vali weapons just encourage it.

The purpose of this PR isn't to "make more melee marines," simply to add more melee. The loadout is the marine's choice, and therefore their fault if it does not perform well.

**Comment by CCorvet** [comment]
Maybe as a way to balance it, it could significantly less damage 1 tile distance, modestly less damage at 2 tile distance, and full at 3?

That way instead of just spear plus, it rewards spacing and still rewards xeno blood.

Also could instead of injecting chems like other vali weapons, it doing some snowflake interactions like that slow sweep, but maybe also root/pull xenos in closer? Reverse of powerfist.

For balance could also make it a 2 handed weapon to have any range, with one handed just being a vali knife with MAYBE extra armor peon. If so the damage penalties should only apply when wielded.

Get over here sound effect when this ability is used mandatory.

**Comment by zzurge** [comment]
No scythe,i cri

**Review by TiviPlus — APPROVED** [review]
get kuro approval
sidenote I see a bunch of new created and modified sprites in icondiff dunno if thats intended

**Comment by theta891** [comment]
> The amount of melee marines I've seen recently are ridiculous. The amount of melee tools there are is ridiculous. The more melee tools there are, the more melee marines there will be, the more marines will just throw. Getting stared at by 4 marines who dont bring a gun isnt fun and I feel like more vali weapons just encourage it.

vali is easy to counter just stop being a dumbass 

so many xenos feed vali users endless blood thats not how you deal with them

grow a brain xeno main

**Comment by ClosetedSkeletonGH** [comment]
> vali is easy to counter just stop being a dumbass

so many xenos feed vali users endless blood thats not how you deal with them

grow a brain xeno main

you have 0 reading comprehension. nowhere in this is it saying melee weapons are good. its me saying theyre overbloated and cause marines to throw, along with in general being a theme conflict and are unfun to go against (easy stomps cause bad players)

**Comment by theta891** [comment]
> you have 0 reading comprehension. nowhere in this is it saying melee weapons are good. its me saying theyre overbloated and cause marines to throw, along with in general being a theme conflict and are unfun to go against (easy stomps cause bad players) or have on your team

if you think more melee tools is causing marines to throw you dont understand why marines loose

**Review by ivanmixo — COMMENTED** [review]
You didn't put it in a vendor from what I can tell, probably not intended.

**Comment by cheapcents** [comment]
> Yeah sorry I was a bit lazy with the sprites, not to mention that I don't know where I was going sprite-wise with "harvester flail"

You could've asked the guy who made the vali sprites to help you

**Pull request closed without merging**